src/lifecycles/load.js

```diff
@@ -18,7 +20,7 @@ export function toLoadPromise(app) {
       const loadPromise = app.loadImpl(getProps(app));
       if (!smellsLikeAPromise(loadPromise)) {
         // The name of the app will be prepended to this error message inside of the handleAppError function
-        throw Error(`single-spa loading function did not return a promise. Check the second argument to registerApplication('${app.name}', loadingFunction, activityFunction)`);
+        throw new InternalError(`single-spa loading function did not return a promise. Check the second argument to registerApplication('${app.name}', loadingFunction, activityFunction)`);
```

Comment:
hmm this looks wrong to me. We're inside of an `if` statement for when the loading function did not return a promise. Not for when the loading function's returned promise failed. Don't we want LOAD_ERROR to happen as a result of a failed network request? Not as the result of a bug in someone's loading function?

I might be misunderstanding the intent of LOAD_ERROR

Reply:
I maybe should change the name from `InternalError`, because really this is signifying that it is an external error that the user didn't provide the right stuff. See line https://github.com/CanopyTax/single-spa/pull/342/files#diff-17157b1bdf3ca457e215ece5f66f5419R69

Reply:
Ah I see I misread the ternary on line 69. Thought that InternalError resulted in load error.

Yeah maybe let's rename `InternalError` to `ValidationError`?